panther_gpiod/CMakeLists.txt

```diff
@@ -0,0 +1,94 @@
+cmake_minimum_required(VERSION 3.10.2)
+project(panther_gpiod)
+
+# Default to C++17
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
+
+# install libgpiod
+include(ExternalProject)
+
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/libgpiod-build/include)
+
+ExternalProject_Add(libgpiod-build
+  GIT_REPOSITORY git://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git
+  GIT_TAG v2.0.2
+  PREFIX ${CMAKE_CURRENT_BINARY_DIR}/libgpiod-build
+  CONFIGURE_COMMAND <SOURCE_DIR>/autogen.sh --prefix=<INSTALL_DIR> --enable-tools=no --enable-bindings-cxx
+  BUILD_COMMAND make -j ${N}
+  INSTALL_COMMAND make install INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/libgpiod-build
+  BUILD_IN_SOURCE 1
+)
```

Comment:
I dedicated a whole day to it, and it's not functioning as expected. The same issue seems linked to a warning during the build process. The library relinks, but without that step, it doesn't work. Since it only impacts the build process, I'd like to leave this topic for a better night's sleep :p

Reply:
I committed several changes, but in my opinion, the progress made is insufficient. Although there was a decrease in the build time, it still took approximately 40-60 seconds to build on the Raspberry Pi